lib/spack/spack/util/prefix.py

```diff
@@ -55,3 +55,6 @@ class Prefix(str):
     """
     def __getattr__(self, attr):
         return Prefix(os.path.join(self, attr))
+
+    def join(self, string):
```

Comment:
Can you add a docstring to this? Something like:
```python
"""Concatenates a string to a prefix.

Parameters:
    string (str): the string to append to the prefix

Returns:
    Prefix: the newly created installation prefix
"""
```
It would also be good to add an example use case of this to the docstring for `Prefix` above.

Reply:
done.